diff --git a/lib/linter/linter.js b/lib/linter/linter.js
index f897b8ddb8c011e53885d6e9a272cf54b3b3e77a..ef38bab21ddd77fcc9f765dd99f0d19d99b6cecd 100644
--- a/lib/linter/linter.js
+++ b/lib/linter/linter.js
@@ -41,7 +41,7 @@ const { getRuleFromConfig } = require("../config/flat-config-helpers");
 const { FlatConfigArray } = require("../config/flat-config-array");
 
 const debug = require("debug")("eslint:linter");
-const MAX_AUTOFIX_PASSES = 10;
+const MAX_AUTOFIX_PASSES = Number.MAX_VALUE;
 const DEFAULT_PARSER_NAME = "espree";
 const DEFAULT_ECMA_VERSION = 5;
 const commentParser = new ConfigCommentParser();